react/SellerWrapper.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useCallback, useEffect, useRef } from 'react'
+import { useProduct, useProductDispatch } from 'vtex.product-context'
+import { useCssHandles } from 'vtex.css-handles'
+import type { Item } from 'vtex.product-context/react/ProductTypes'
+
+import useSeller from './hooks/useSeller'
+
+const CSS_HANDLES = ['sellerWrapper']
+const SELECT_ITEM_EVENT = 'SET_SELECTED_ITEM'
+
+type SellerWrapperProps = {
+  children: React.ReactNode
+}
+
+const SellerWrapper = ({ children }: SellerWrapperProps) => {
+  const { seller } = useSeller()
+  const dispatch = useProductDispatch()
+  const { selectedItem, product } = useProduct() ?? {}
+  const latestItem = useRef((null as unknown) as Item)
+  const handles = useCssHandles(CSS_HANDLES)
+
+  const memoizedCallback = useCallback(
+    itemSeller => ({
+      ...itemSeller,
+      sellerDefault: itemSeller.sellerId === seller,
+    }),
+    [seller]
+  )
+
+  const newCurrentSelectedItem = product?.items?.find(item =>
+    item.sellers?.find(
+      itemSeller =>
+        itemSeller.sellerId === seller &&
+        itemSeller.commertialOffer.AvailableQuantity > 0
+    )
+  )
+
+  useEffect(() => {
+    if (!seller || !newCurrentSelectedItem || !selectedItem || !dispatch) return
+
+    const { sellers } = newCurrentSelectedItem
+
+    dispatch?.({
+      type: SELECT_ITEM_EVENT,
+      args: {
+        item: {
+          ...newCurrentSelectedItem,
+          sellers: sellers.map(memoizedCallback),
+        },
+      },
+    })
+  }, [seller])
+
+  useEffect(() => {
+    if (!seller || !selectedItem) return
+
+    const { sellers } = selectedItem
+
+    if (latestItem.current) {
+      const { sellers: latestSellers } = latestItem.current
+
+      if (JSON.stringify(sellers) === JSON.stringify(latestSellers)) return
+    }
+
+    const newItem = {
+      ...selectedItem,
+      sellers: sellers.map(memoizedCallback),
+    }
+
+    dispatch?.({
```

Comment:
There are 2 effects dispatching this event, and they can be dispatched together, with this one probably winning, being the last one. Ideally we'd only dispatch a single event here though, right? Or do we need to dispatch two separate events? Can't we unite these logics into a single effect hook that decides which of the two payloads to send?